### PR TITLE
chore: Move Windows tests to GHA custom runner

### PR DESCRIPTION
**Pull request by jszwedko**
Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: tests/crash.rs [hunk 1]
```diff
@@ -68,6 +68,7 @@ impl Sink<Event> for PanicSink {
 }
 
 #[tokio::test]
+#[ignore]
 async fn test_sink_panic() {
```
Comment (spencergilbert) [inline tests/crash.rs:72]: This is a `todo` fix soon yea?

**Review by spencergilbert — APPROVED** [review]

**Comment by pablosichert** [comment]
Is it intentional to move to Windows 2022 here? I assume that customers on Windows might predominantly not use the latest versions 😄

**Comment by binarylogic** [comment]
Do we want to move our Windows build workflows as well?

File: tests/crash.rs (hunk 1 above)
Comment (tobz) [reply tests/crash.rs:72]: Ye.

**Comment by tobz** [comment]
Hmmm, yeah, I hadn't thought of the release process.  Let me move this back to 2019 and also align the release builder to use these new runners as well, just so we know they're at parity.

I don't think 2019 vs 2022 actually matters in this case, but there's no harm staying on it, really.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -174,9 +174,10 @@ jobs:
           path: "./target/artifacts/vector-${{ env.VECTOR_VERSION }}-x86_64-apple-darwin.tar.gz"
 
   build-x86_64-pc-windows-msvc-packages:
-    runs-on: [self-hosted, windows, x64, general]
+    runs-on: [windows, release-builder]
```
Comment (jszwedko) [inline .github/workflows/release.yml:177]: I think you have to update `nightly.yml` too.

Comment (tobz) [reply .github/workflows/release.yml:177]: Good call.

**Review by jszwedko — COMMENTED** [review]
I can't approve since I opened the PR but ✅

**Review by pablosichert — APPROVED** [review]

**Pull request merged**